mmdet/apis/ote/apis/detection/ote_utils/dataset.py

```diff
@@ -0,0 +1,310 @@
+# Copyright (C) 2021 Intel Corporation
```

Comment:
I would not depend on neither `mmdet` nor `mmcv` ([like here](https://github.com/openvinotoolkit/training_extensions/blob/brave_new_world/ote_cli/ote_cli/datasets/object_detection/dataset.py)) and renamed `MMDataset` to `ObjectDetectionDataset` and put it to new SDK

Reply:
As someone said that it is possible to make `Dataset` not abstract, probably some stuff will be easier

Reply:
Of course implementation should not depend on `mmdet` and `mmcv` to be moved to the SDK.

The opens are:

- Which of these implementations should we base off of then?
- Should we refactor the Dataset to be easily adapted for other tasks as well?